src/transport/socket.c

```diff
@@ -112,6 +113,24 @@ socket_parse_uri(const char *uri_string)
 	}
 
 	if (!g_strcmp0(uri.scheme, "tcp")) {
+		if (uri.host[0] == '[') {
+			uri.port = 0;
+			ptr = strchr(uri_string, ']');
+			if (ptr) {
+				ptr += 1;
+				if (*ptr == ':') {
+					uri.port = atoi(ptr+1);
+				}
+			}
+			if (ptr == NULL || uri.port == 0) {
+				g_free(uri_copy);
+				rpc_set_last_errorf(EINVAL,
+				    "Missing tcp port");
+					return (NULL);
+			}
+			uri.host = "::";
```

Comment:
don't we want to extract the IPv6 address and not just assume it's localhost? I still want to be able to connect with `tcp://[fe80::4fb4:9957:6a93:4421]:5001`